src/frontend/src/handler/alter_table_with_sr.rs

```diff
@@ -47,35 +45,31 @@ pub async fn handle_refresh_schema(
         bail_not_implemented!("alter table with incoming sinks");
     }
 
-    let format_encode = {
-        let format_encode = get_format_encode_from_table(&original_table)?;
-        if !format_encode
-            .as_ref()
-            .is_some_and(schema_has_schema_registry)
-        {
-            return Err(ErrorCode::NotSupported(
-                "tables without schema registry cannot refreshed".to_owned(),
-                "try `ALTER TABLE .. ADD/DROP COLUMN ...` instead".to_owned(),
-            )
-            .into());
-        }
-        format_encode.unwrap()
-    };
-
-    // NOTE(st1page): since we have not implemented alter format encode for table, it is actually no use.
-    let definition = alter_definition_format_encode(
-        &original_table.definition,
-        format_encode.row_options.clone(),
-    )?;
+    let format_encode = get_format_encode_from_table(&original_table)?;
+    if !format_encode
+        .as_ref()
+        .is_some_and(schema_has_schema_registry)
+    {
+        return Err(ErrorCode::NotSupported(
+            "tables without schema registry cannot be refreshed".to_owned(),
+            "try `ALTER TABLE .. ADD/DROP COLUMN ...` instead".to_owned(),
+        )
+        .into());
+    }
 
-    let [definition]: [_; 1] = Parser::parse_sql(&definition)
-        .context("unable to parse original table definition")?
-        .try_into()
-        .unwrap();
+    let definition = original_table
+        .create_sql_ast_purified()
+        .context("unable to parse original table definition")?;
 
     let (source, table, graph, col_index_mapping, job_type) = {
-        let result =
-            get_replace_table_plan(&session, table_name, definition, &original_table).await;
+        let result = get_replace_table_plan(
+            &session,
+            table_name,
+            definition,
+            &original_table,
+            SqlColumnStrategy::Ignore,
```

Comment:
Resolve columns from schema registry again and fully replace the current definition.